File: src/utils/formatters.ts
/**
 * Format countdown time from seconds to a human readable string
 */
export function formatCountdown(timeUntil: number): string {
  const days = Math.floor(timeUntil / (24 * 60 * 60))
  const hours = Math.floor((timeUntil % (24 * 60 * 60)) / (60 * 60))
  const minutes = Math.floor((timeUntil % (60 * 60)) / 60)

  let timeString = ''
  if (days > 0) timeString += `${days} day${days > 1 ? 's' : ''} `
  if (hours > 0) timeString += `${hours} hour${hours > 1 ? 's' : ''} `
  if (minutes > 0) timeString += `${minutes} minute${minutes > 1 ? 's' : ''}`

  return timeString.trim()
}

/**
 * Format a date to user-friendly format (Discord timestamp)
 * Format: <t:timestamp:D> - Shows date only (e.g., "December 25, 2023")
 */
export function formatDate(date: Date): string {
  return `<t:${Math.floor(date.getTime() / 1000)}:D>`
}

/**
 * Format time to user-friendly format (Discord timestamp)
 * Format: <t:timestamp:t> - Shows time only (e.g., "3:30 PM")
 */
export function formatTime(date: Date): string {
  return `<t:${Math.floor(date.getTime() / 1000)}:t>`
}

/**
 * Format date and time for air date display (Discord timestamp)
 * Format: <t:timestamp:F> - Full date and time (e.g., "Monday, December 25, 2023 3:30 PM")
 */
export function formatAirDate(date: Date): string {
  return `<t:${Math.floor(date.getTime() / 1000)}:F>`
}

/**
 * Format compact date and time for lists (Discord timestamp)
 * Format: <t:timestamp:f> - Short date and time (e.g., "December 25, 2023 3:30 PM")
 */
export function formatCompactDateTime(date: Date): string {
  return `<t:${Math.floor(date.getTime() / 1000)}:f>`
}

/**
 * Format a time as Discord relative timestamp
 * Format: <t:timestamp:R> - Relative time (e.g., "in 2 hours", "3 days ago")
 */
export function formatRelativeTimestamp(date: Date): string {
  return `<t:${Math.floor(date.getTime() / 1000)}:R>`
}

File: go/internal/utils/formatters.go
package utils

import (
	"fmt"
	"time"
)

// FormatCountdown formats countdown time from seconds to a human readable string
func FormatCountdown(timeUntil int) string {
	days := timeUntil / (24 * 60 * 60)
	hours := (timeUntil % (24 * 60 * 60)) / (60 * 60)
	minutes := (timeUntil % (60 * 60)) / 60

	var timeString string
	if days > 0 {
		if days > 1 {
			timeString += fmt.Sprintf("%d days ", days)
		} else {
			timeString += fmt.Sprintf("%d day ", days)
		}
	}
	if hours > 0 {
		if hours > 1 {
			timeString += fmt.Sprintf("%d hours ", hours)
		} else {
			timeString += fmt.Sprintf("%d hour ", hours)
		}
	}
	if minutes > 0 {
		if minutes > 1 {
			timeString += fmt.Sprintf("%d minutes", minutes)
		} else {
			timeString += fmt.Sprintf("%d minute", minutes)
		}
	}

	if timeString == "" {
		return "less than a minute"
	}

	return timeString
}

// FormatDate formats a date to user-friendly format
func FormatDate(date time.Time) string {
	return date.Format("Monday, January 2, 2006")
}

// FormatTime formats time to user-friendly format (12-hour)
func FormatTime(date time.Time) string {
	return date.Format("3:04 PM")
}

// FormatAirDate formats date and time for air date display
func FormatAirDate(date time.Time) string {
	return fmt.Sprintf("%s at %s", FormatDate(date), FormatTime(date))
}

// FormatCompactDateTime formats compact date and time for lists
func FormatCompactDateTime(date time.Time) string {
	return date.Format("Jan 2, 3:04 PM")
}
